### DISCUSSION TITLE
Disable `CSharpUsing/PreferQualifiedReference` code rule

### DISCUSSION THREAD

**Discussion by moreal** [planetarium/NineChronicles.Standalone#2637] [General] [answered]
#2607 에서 설정된 `NineChronicles.Headless.Executable.sln.DotSettings` 파일 내 `/Default/CodeStyle/CSharpUsing/PreferQualifiedReference/@EntryValue` 의 값을 `False` 로 변경하면 어떨까요?

불편함을 겪은 상황을 예로 들면, Rider IDE에서 사용할 때 `Serializable`에 필요한 using 문을 자동으로 기입해주는 기능을 사용하는데, 이 값이 `True` 임으로 인해서 파일 상단에 `using System;` 라인을 추가하는 대신 `System.SerializableAttribute` 같은 꼴로 변경됩니다.

**Comment by moreal** [comment]
@planetarium/9c-backend 팀과 관련 룰 추가해주신 @eugene-doobu 님 의견 부탁드립니다 🙏🏻

**Reply by eugene-doobu** [reply]
관련해서 해당 옵션을 셋팅했던 이유와 해당 값을 설정하니 작업에 불편하여 False로 설정한 과정이 아래 pr에 담겨있습니다! 해당 옵션을 적용했던 프로젝트에 일괄적으로 클라이언트와 동일하게 옵션을 변경하려고 했는데 미루다가 설정 하지 못했네요, 죄송합니다.
저도 해당 옵션 적용하는 것 찬성합니다!

https://github.com/planetarium/NineChronicles/pull/6313/files

**Reply by eugene-doobu** [reply]
- lib9c/headless등에 적용된 StyleCop 코드 규칙에 맞게 코드 변경을 진행 후 추가적으로 수정되는 스타일입니다.
- StyleCop 규칙에서 using문에 namespace를 작성할 때 무조건 namespace의 풀 네임을 적용해야 하는 규칙이 있습니다
- 이를 대비하기 위해 라이더에서 'PreferQualifiedReference' 옵션을 True로 설정해두었는데 이로 인해 필드의 클래스를 지정할때에도 자동적으로 full namespace가 들어가는 케이스가 있고 이게 신경쓰이는 상황이 반복되어 False로 옵션을 변경해두었습니다

```
SA1210: Using directives should be ordered alphabetically by the namespaces.
SA1135: Using directive for namespace 'Lib9c.Tests.Action' should be qualified
```

**Reply by moreal** [reply, marked as answer]
https://github.com/planetarium/NineChronicles.Headless/pull/2639 에서 해결되었습니다!
